x-pack/plugins/upgrade_assistant/server/lib/es_deprecation_logging_apis.ts

```diff
@@ -30,17 +28,38 @@ export async function setDeprecationLogging(
 ): Promise<DeprecationLoggingStatus> {
   const { body: response } = await dataClient.asCurrentUser.cluster.putSettings({
     body: {
+      persistent: {
+        'logger.deprecation': isEnabled ? 'WARN' : 'ERROR',
+        'cluster.deprecation_indexing.enabled': isEnabled,
+      },
+      /*
+       * If we only set the persistent setting, we can end up in a situation in which a user has
+       * set transient on/off. And when toggling and reloading the page the transient setting will
+       * have priority over it thus "overriding" whatever the user selected.
+       */
       transient: {
         'logger.deprecation': isEnabled ? 'WARN' : 'ERROR',
+        'cluster.deprecation_indexing.enabled': isEnabled,
       },
     },
   });
 
   return {
-    isEnabled: isDeprecationLoggingEnabled(response),
+    isEnabled,
+    isLoggerDeprecationEnabled: isDeprecationLoggingEnabled(response),
   };
 }
 
+export function isClusterDeprecationLoggingEnabled(settings: any) {
```

Comment:
I'm confused by this name. It makes me ask "What is _cluster_ deprecation logging and how does it compare to the standard deprecation logging?" But this is asking the wrong question, since this function determines whether logs will be indexed. In which case, I think a clearer name would be `isDeprecationLogIndexingEnabled`.

Reply:
Nice catch!